timely/src/dataflow/operators/core/capture/capture.rs

```diff
@@ -79,30 +79,26 @@ pub trait Capture<T: Timestamp, C: Container + Data> {
     /// let (send0, recv0) = ::std::sync::mpsc::channel();
     /// let send0 = Arc::new(Mutex::new(send0));
     ///
-    /// timely::execute(timely::Config::thread(), move |worker| {
-    ///
-    ///     // this is only to validate the output.
-    ///     let send0 = send0.lock().unwrap().clone();
-    ///
-    ///     // these allow us to capture / replay a timely stream.
-    ///     let list = TcpListener::bind("127.0.0.1:8001").unwrap();
-    ///     let send = TcpStream::connect("127.0.0.1:8001").unwrap();
-    ///     let recv = list.incoming().next().unwrap().unwrap();
-    ///
-    ///     recv.set_nonblocking(true).unwrap();
+    /// // these allow us to capture / replay a timely stream.
+    /// let list = TcpListener::bind("127.0.0.1:8001").unwrap();
+    /// let send = TcpStream::connect("127.0.0.1:8001").unwrap();
+    /// let recv = list.incoming().next().unwrap().unwrap();
+    /// recv.set_nonblocking(true).unwrap();
     ///
-    ///     worker.dataflow::<u64,_,_>(|scope1|
+    /// std::thread::scope(move |s| {
+    ///     s.spawn(move || timely::example(move |scope1| {
```

Comment:
I moved the two sides in separate threads since if one operator blocks while writing to the socket nothing will unblock it